src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,163 @@
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.USER: {
+          const rawUser = data.data.resolved.users[o.value];
+          const rawMember = data.data.resolved.members[o.value];
+          if (rawMember) {
+            rawMember.user = rawUser;
+          }
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            user: this.client.users ? this.client.users.add(rawUser, false) : new User(this.client, rawUser),
+            member: rawMember ? this.guild?.members.add(rawMember, false) ?? null : null,
+          };
+        }
+        case ApplicationCommandOptionType.CHANNEL: {
+          const raw = data.data.resolved.channels[o.value];
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            channel: this.client.channels
+              ? this.client.channels.add(raw, false)
+              : Channel.create(this.client, raw, this.guild),
+          };
+        }
+        case ApplicationCommandOptionType.ROLE: {
+          const raw = data.data.resolved.roles[o.value];
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            role: this.guild ? this.guild.roles.add(raw, false) : new Role(this.client, raw, undefined),
+          };
+        }
+        default:
+          return o;
+      }
+    };
+    /**
+     * The options passed to the command.
+     * @type {Object}
+     * @readonly
+     */
+    this.options = data.data.options.map(map);
+  }
+
+  /**
+   * The timestamp the interaction was created at.
+   * @type {number}
+   * @readonly
+   */
+  get createdTimestamp() {
+    return Snowflake.deconstruct(this.id).timestamp;
+  }
+
+  /**
+   * The time the interaction was created at.
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * Acknowledge this interaction without content.
+   * @param {Object} [options] Options
+   */
+  async acknowledge(options = {}) {
+    await this.syncHandle.acknowledge(options);
+  }
+
+  /**
+   * Reply to this interaction.
+   * @param {(StringResolvable | APIMessage)?} content The content for the message.
+   * @param {(MessageOptions | MessageAdditions)?} options The options to provide.
+   */
+  async reply(content, options) {
+    let apiMessage;
+
+    if (content instanceof APIMessage) {
+      apiMessage = content.resolveData();
+    } else {
+      apiMessage = APIMessage.create(this, content, options).resolveData();
+    }
+
+    const resolved = await apiMessage.resolveFiles();
+
+    if (!this.syncHandle.reply(resolved)) {
+      await this.client.api.webhooks(this.applicationID, this.token).post({
+        auth: false,
+        data: resolved.data,
+        files: resolved.files,
+        query: { wait: true },
```

Comment:
wait is always true on this endpoint. I could see this being there as a failsafe if that were to change, just wanted to point it out though.